src/test/java/uk/ac/cam/cl/dtg/isaac/dos/eventbookings/PgEventBookingsTest.java

```diff
@@ -100,4 +109,61 @@ void getEventBookingStatusCounts_checkEventBookingStatusCounts_canCopeWithCompli
     assertEquals(expectedStatusCounts, actualStatusCounts, "Every row should be represented in the result");
     verify(mockedObjects);
   }
+
+  @Test
+  void testFindAllByEventIds() throws SQLException, SegueDatabaseException {
+    List<String> eventIds = Arrays.asList("event1", "event2");
+    String expectedQuery = "SELECT event_bookings.* FROM event_bookings JOIN users ON users.id=user_id WHERE event_id IN (?, ?) AND NOT users.deleted";
+    Instant now = now();
+
+    EventBooking expectedBooking1 =
+        new PgEventBooking(1L, 1L, 7L, "event1", BookingStatus.CONFIRMED, now, now, null);
+    EventBooking expectedBooking2 =
+        new PgEventBooking(2L, 2L, 7L, "event2", BookingStatus.CONFIRMED, now, now, null);
+
+    expect(dummyPostgresSqlDb.getDatabaseConnection()).andReturn(dummyConnection);
+    expect(dummyConnection.prepareStatement(expectedQuery)).andReturn(dummyPreparedStatement);
+    dummyPreparedStatement.setString(1, "event1");
+    dummyPreparedStatement.setString(2, "event2");
+    expect(dummyPreparedStatement.executeQuery()).andReturn(dummyResultSet);
+
+    expect(dummyResultSet.next()).andReturn(true);
+    expect(dummyResultSet.getLong("id")).andReturn(1L);
+    expect(dummyResultSet.getLong("user_id")).andReturn(1L);
+    expect(dummyResultSet.getLong("reserved_by")).andReturn(7L);
+    expect(dummyResultSet.getString("event_id")).andReturn("event1");
+    expect(dummyResultSet.getString("status")).andReturn("CONFIRMED");
+    expect(dummyResultSet.getTimestamp("created")).andReturn(Timestamp.from(now));
+    expect(dummyResultSet.getTimestamp("updated")).andReturn(Timestamp.from(now));
+    expect(dummyResultSet.getObject("additional_booking_information")).andReturn(null);
+
+    expect(dummyResultSet.next()).andReturn(true);
+    expect(dummyResultSet.getLong("id")).andReturn(2L);
+    expect(dummyResultSet.getLong("user_id")).andReturn(2L);
+    expect(dummyResultSet.getLong("reserved_by")).andReturn(7L);
+    expect(dummyResultSet.getString("event_id")).andReturn("event2");
+    expect(dummyResultSet.getString("status")).andReturn("CONFIRMED");
+    expect(dummyResultSet.getTimestamp("created")).andReturn(Timestamp.from(now));
+    expect(dummyResultSet.getTimestamp("updated")).andReturn(Timestamp.from(now));
+    expect(dummyResultSet.getObject("additional_booking_information")).andReturn(null);
+
+    expect(dummyResultSet.next()).andReturn(false);
+    dummyResultSet.close();
+    dummyPreparedStatement.close();
+    dummyConnection.close();
+
+    // Create an array of all mocked objects
+    Object[] mockedObjects = {dummyPostgresSqlDb, dummyObjectMapper, dummyConnection, dummyPreparedStatement, dummyResultSet};
+    replay(mockedObjects);
+
+    PgEventBookings pgEventBookings = this.buildPgEventBookings();
+    Iterable<EventBooking> result = pgEventBookings.findAllByEventIds(eventIds);
+    assertNotNull(result);
+
+    List<EventBooking> resultList = (List<EventBooking>) result;
```

Comment:
I think that this casting is only working because you're actually returning a `List`. In other words, this test is now tied to the implementation of the function that you're testing, when it should be completely agnostic. One way to test this properly is by having multiple assertions. For instance, you can check that the result size is exactly 2. Then you can use streams to find a result with ID = 1, and then you do a deep comparison for that object in particular. And then you do the same thing for the result with ID = 2. By doing it this way, you don't care how the function under test has been implemented. You test the functionality, rather than the implementation. Does that make sense?